File: server/models/FeaturedSection.js
import mongoose from "mongoose";

const SECTION_KEYS = [
  "flash-sale",
  "trending",
  "new-arrivals",
  "best-sellers",
  "recommended"
];

const featuredSectionSchema = new mongoose.Schema(
  {
    key: { type: String, required: true, unique: true, enum: SECTION_KEYS },
    title: { type: String, required: true, trim: true, maxlength: 80 },
    products: [{ type: mongoose.Schema.Types.ObjectId, ref: "Product" }],
    sortOrder: { type: Number, default: 0 },
    countdownStartsAt: { type: Date, default: null },
    countdownEndsAt: { type: Date, default: null },
    isActive: { type: Boolean, default: true }
  },
  { timestamps: true }
);

export { SECTION_KEYS };
export default mongoose.model("FeaturedSection", featuredSectionSchema);

File: ridercraft_mobile/lib/models/featured_section.dart
import 'product.dart';

/// Featured product section from `GET /featured-sections`.
///
/// Keys: flash-sale, trending, new-arrivals, best-sellers, recommended.
class FeaturedSection {
  final String id;
  final String key;
  final String title;
  final List<Product> products;
  final DateTime? countdownStartsAt;
  final DateTime? countdownEndsAt;
  final bool isActive;

  const FeaturedSection({
    required this.id,
    required this.key,
    required this.title,
    this.products = const [],
    this.countdownStartsAt,
    this.countdownEndsAt,
    this.isActive = true,
  });

  factory FeaturedSection.fromJson(Map<String, dynamic> json) =>
      FeaturedSection(
        id: (json['_id'] ?? '') as String,
        key: (json['key'] ?? '') as String,
        title: (json['title'] ?? '') as String,
        products: (json['products'] as List?)
                ?.map((e) => Product.fromJson(e as Map<String, dynamic>))
                .toList() ??
            const [],
        countdownStartsAt: json['countdownStartsAt'] != null
            ? DateTime.tryParse(json['countdownStartsAt'] as String)
            : null,
        countdownEndsAt: json['countdownEndsAt'] != null
            ? DateTime.tryParse(json['countdownEndsAt'] as String)
            : null,
        isActive: (json['isActive'] ?? true) as bool,
      );
}
